src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -74,27 +74,27 @@ void setUserStore(UserStore userStore) {
     public void doGet(HttpServletRequest request, HttpServletResponse response)
             throws IOException, ServletException {
 
-        String username = request.getParameter("username");
+        String requestUrl = request.getRequestURI();
+        String username = requestUrl.substring("/users/".length());
 
-        // if user doesn't exist, set "username" attribute as empty string
-        // so that profile.jsp sends to a page that states that user doesn't exist
-        // otherwise set the username attribute
+        // see if user exists
         User user = userStore.getUser(username);
+
         if (user == null) {
-            request.setAttribute("username", "");
+            System.out.println("No user exists.");
```

Comment:
Where will this print statement be seen if this condition is true? Will it pop up on the webpage?

Reply:
I changed it so it will print out in the terminal, but before I had set the attribute "username" to be an empty string to send to profile.jsp to display on the web page for the user to see "No user exists". Should I revert to that since it makes it easier for the user to see?

Reply:
Great question! No it won't show up on the page, remind me during our meeting next tuesday to show you where to view such messages. And BTW we should use loggers for debug messages instead of println().

Reply:
Hmm, I think that maybe we should have a "page doesn't exist" or "page not found" when users search for a page that isn't there. I'm not sure how we would map that though.

Reply:
Sounds good! @ppbonds good idea, I made a web page that shows "User does not exist" in that case in profile.jsp